frappe/model/db_query.py

```diff
@@ -871,7 +871,7 @@ def prepare_filter_condition(self, ft: FilterTuple) -> str:
 					value = value.replace("\\", "\\\\").replace("%", "%%")
 
 			elif f.operator == "=" and df and df.fieldtype in ["Link", "Data"]:  # TODO: Refactor if possible
```

Comment:
```suggestion
			elif f.operator == "=" and df and df.fieldtype in ["Link", "Data"]:
```

This is present because the condition above wild-card applies to all possible non-numeric types. :hankey: